src/elf_sections.rs

```diff
@@ -82,17 +82,34 @@ impl Iterator for ElfSectionIter {
 pub struct ElfSection {
     name_index: u32,
     typ: u32,
-    pub flags: u64,
-    pub addr: u64,
+    flags: u64,
+    addr: u64,
     offset: u64,
-    pub size: u64,
+    size: u64,
     link: u32,
     info: u32,
     addralign: u64,
     entry_size: u64,
 }
 
 impl ElfSection {
+    pub fn section_type(&self) -> ElfSectionType {
+        match self.typ {
+            0 => ElfSectionType::Unused,
+            1 => ElfSectionType::ProgramSection,
+            2 => ElfSectionType::LinkerSymbolTable,
+            3 => ElfSectionType::StringTable,
+            4 => ElfSectionType::RelaRelocation,
+            5 => ElfSectionType::SymbolHashTable,
+            6 => ElfSectionType::DynamicLinkingTable,
+            7 => ElfSectionType::Note,
+            8 => ElfSectionType::Uninitialized,
+            9 => ElfSectionType::RelRelocation,
+            10 => ElfSectionType::Reserved,
+            11 => ElfSectionType::DynamicLoaderSymbolTable,
+            _ => panic!(),
```

Comment:
Perhaps it would not be best to panic here because environment and processor specific flags are allowed.